src/solvers/dg_curved/dg_2d.jl

```diff
@@ -77,23 +88,25 @@ function calc_interface_flux!(u::AbstractArray{<:Any,4}, mesh::CurvedMesh{2},
     # Interfaces in x-direction (`orientation` = 1)
     calc_interface_flux!(elements.surface_flux_values,
                          elements.left_neighbors[1, element],
-                         element, 1, u, equations, dg, cache)
+                         element, 1, u, mesh, equations, dg, cache)
 
     # Interfaces in y-direction (`orientation` = 2)
     calc_interface_flux!(elements.surface_flux_values,
                          elements.left_neighbors[2, element],
-                         element, 2, u, equations, dg, cache)
+                         element, 2, u, mesh, equations, dg, cache)
   end
 
   return nothing
 end
 
 
 @inline function calc_interface_flux!(surface_flux_values, left_element, right_element,
-                                      orientation, u, equations, dg::DG, cache)
+                                      orientation, u,
+                                      mesh::CurvedMesh{2}, equations,
+                                      dg::DG, cache)
   # This is slow for LSA, but for some reason faster for Euler (see #519)
   if left_element <= 0 # left_element = 0 at boundaries
-    return surface_flux_values
+    return nothing
```

Comment:
Hmm, good point. Currently, we just `return nothing` from most (all?) functions called inside of `rhs!`, often implicitly by having a `for` loop as last statement in the function body. Moreover, we never use the return value and do not document it anywhere, so this is not breaking. Thus, I wanted to achieve consistency within Trixi.
However, you're totally right in the sense that we should follow the Julia guidelines and provide predictable return values. On the other hand, some cases are not really clear - shall we also return the `cache` if it is modified? What if the `cache` is the first argument? What do you think, @sloede?

Reply:
I'm torn here. My personal heuristic has been: If we're using _lightweight_, "Julian" functions (i.e., small functions with hopefully no side effects and a single, specific purpose), the modified argument should be returned. Specifically, if the functions are designed to be used (or to be usable at lest) in different circumstances.

However, many high-level functions called from `rhs!` modify more than one argument, and they do not make much sense to be callend anywhere else, since they rely on a rich setup (e.g., here, `cache` needs to have `elements` which have `contravariant_vectors`, `u` and `surface_flux_values` need to be of very specific dimensions that need to match `equations` and `u` etc.). Thus for these _rich_ functions I'd actually favor returning `nothing`, since they are not meant to be used in a chained way. In addition, this way it is also more convenient when testing them manually, since when called on the REPL they don't show anything.

Reply:
Okay, that sounds reasonable to me :+1:

Reply:
@erik-f Does this resolve the issue for you as well?